---

- name: GitLab post | check to see if SSL cert already applied
  become: false
  get_certificate:
    host: "gitlab.{{ec2_name_prefix|lower}}.{{workshop_dns_zone}}"
    port: 443
  delegate_to: localhost
  run_once: true
  failed_when: false
  register: check_cert

- block:
    - name: GitLab post | stop gitlab
      command: gitlab-ctl stop

    - name: GitLab post | Install EPEL
      dnf:
        name: "https://dl.fedoraproject.org/pub/epel/epel-release-latest-8.noarch.rpm"
        state: present

    # temporary fix for rhel 8.2 dnf substitution
    - name: fix EPEL repo substitution
      replace:
        path: "{{ item }}"
        regexp: '\$releasever'
        replace: '8'
      loop:
        - /etc/yum.repos.d/epel-modular.repo
        - /etc/yum.repos.d/epel.repo
      when:
        - ansible_distribution_major_version|int == 8

    - name: GitLab post | IInstall base packages
      dnf:
        name:
          - python3-pip
          - python3-devel

    # directions found here https://certbot.eff.org/lets-encrypt/centosrhel8-other
    - name: Download and install certbot
      get_url:
        url: https://dl.eff.org/certbot-auto
        dest: /usr/local/bin/certbot-auto
        mode: '0755'
        owner: "root"

    - name: GitLab post | issue cert
      shell: /usr/local/bin/certbot-auto certonly --no-bootstrap --standalone -d gitlab.{{ec2_name_prefix|lower}}.{{workshop_dns_zone}} --email ansible-network@redhat.com --noninteractive --agree-tos
      register: issue_cert
      until: issue_cert is not failed
      retries: 5
      ignore_errors: true

    - name: GitLab post | reconfigure gitlab
      command: gitlab-ctl reconfigure

    - name: GitLab post | start gitlab
      command: gitlab-ctl start

    - name: GitLab post | wait for gitlab to start
      become: false
      get_certificate:
        host: "gitlab.{{ec2_name_prefix|lower}}.{{workshop_dns_zone}}"
        port: 443
      delegate_to: localhost
      register: check_cert2
      retries: 120
      delay: 1
      until: check_cert2 is not failed

  when: check_cert.cert is not defined
